Feature/add notes

ltwheeler opened the pull request:
Adds a way to store and view notes over time on both a course and instructor.

Adds two input boxes in the assignments view (one for each pivot).
Shows up as a label in print view.

Issue:
https://trello.com/c/9q98Keux/1779-courses-add-a-way-to-store-and-view-notes-about-a-course-or-instructor-over-time

<img width="1249" alt="screen shot 2018-07-31 at 7 48 41 am" src="https://user-images.githubusercontent.com/1776859/43467439-8ec25e2c-9496-11e8-8fad-7a651f019c38.png">
<img width="1240" alt="screen shot 2018-07-31 at 7 48 53 am" src="https://user-images.githubusercontent.com/1776859/43467448-9153e160-9496-11e8-9a39-35fc582dbe81.png">

cthielen commented on app/assignment/directives/instructorAssignmentTable.js:
```diff
@@ -540,6 +562,34 @@ let instructorAssignmentTable = function ($rootScope, AssignmentActionCreators,
+				let $el = $(e.target);
+				if ($el.data('event-type') != 'setInstructorNote') { return; }
+
+				if (e.key == 	"Enter") {
```
Fix spacing after ==

cthielen commented on app/assignment/directives/instructorAssignmentTable.js:
```diff
@@ -540,6 +562,34 @@ let instructorAssignmentTable = function ($rootScope, AssignmentActionCreators,
+			});
+
+			// Handle input box edits
+			element.on("change", function(e) {
```
Does 'change' handle blur? That seems like a logical time to save.

ltwheeler replied:
As I understand it "change" is similar to "blur", but only fires if the value was actually changed, whereas blur would fire regardless.
https://api.jquery.com/change/

cthielen replied:
Cool, thanks!

cthielen commented on app/assignment/services/assignmentActionCreators.js:
```diff
@@ -34,6 +34,45 @@ class AssignmentActionCreators {
+				instructorNote.note = note;
+
+				_self.AssignmentService.updateInstructorNote(instructorNote).then(function (newInstructorNote) {
+					_self.$rootScope.$emit('toast', { message: "Updated course note", type: "SUCCESS" });
```
Should read "Updated instructor note"

cthielen commented on app/assignment/services/assignmentActionCreators.js:
```diff
@@ -34,6 +34,45 @@ class AssignmentActionCreators {
+					};
+					_self.AssignmentStateService.reduce(action);
+				}, function (err) {
+					_self.$rootScope.$emit('toast', { message: "Could not update course note.", type: "ERROR" });
```
Should read 'Could not update instructor note'

cthielen commented on app/assignment/services/assignmentService.js:
```diff
@@ -26,6 +26,12 @@ class AssignmentService {
+				return ApiService.put("/api/courseView/courses/" + course.id, course);
+			},
+			updateInstructorNote: function (instructorNote) {
+				return ApiService.put("/api/schedules/" + instructorNote.scheduleId + "/instructors/" + instructorNote.instructorId + "/instructorNotes", instructorNote);
```
By this API route, are we storing notes per instructor and per schedule?

ltwheeler replied:
That is correct.

ltwheeler commented:
- Fixed private staff comment modal.
- Updated styles to be taller and use all available space.
- Tweaked border style and added a dividing line.
<img width="241" alt="screen shot 2018-08-10 at 7 44 32 am" src="https://user-images.githubusercontent.com/1776859/43964101-f643d3c8-9c70-11e8-917b-043213e79eb6.png">

cthielen commented:
Two issues:

Modal instructor notes 'save' button does not close dialog. (Not sure what it did before but it should close on save.)

and

Instructor unavailability modal does not open. JS error:

[Error] TypeError: scope.openUnavailabilityModal is not a function. (In 'scope.openUnavailabilityModal(instructorId)', 'scope.openUnavailabilityModal' is undefined)
	(anonymous function) (instructorAssignmentTable.js:665)
	dispatch (lib.js:5087)

cthielen commented on app/assignment/css/assignments.css:
```diff
@@ -972,3 +972,26 @@ li.active-term .dropdown-term-row {
+}
+
+.assignment__divider {
+  height: 1px;
```
Inconsistent tabbing

cthielen commented on app/assignment/directives/courseAssignmentTable.js:
```diff
@@ -322,6 +331,30 @@ let courseAssignmentTable = function ($rootScope, AssignmentActionCreators) {
+				let $el = $(e.target);
+				if ($el.data('event-type') != 'setCourseNote') { return; }
+
+				if (e.key == 	"Enter") {
```
Needless extra spaces after ==
